2023-risks/M8: Insufficient Cryptography

```diff
@@ -0,0 +1,106 @@
+
+**Loss of Intellectual Property**: Inadequate cryptography can jeopardize the protection of proprietary algorithms, trade secrets, or other intellectual property embedded within the mobile application. If attackers are able to decrypt and extract this valuable information, it can be exploited for competitive advantage by rival companies or sold on the black market.
+
+**Financial Losses**: Insufficient cryptography can lead to financial losses in multiple ways. For instance, if payment transactions or financial data are improperly encrypted, it can expose customers to fraud and unauthorized access to their funds. Additionally, the costs associated with investigating and remediating security breaches, compensating affected customers, and addressing legal ramifications can be substantial.
+
+**Compliance and Legal Consequences**: Many industries have specific data protection and privacy regulations that mandate the use of strong encryption for sensitive information. Inadequate cryptography can result in non-compliance with these regulations, leading to legal consequences, fines, or sanctions imposed by regulatory authorities.
+
+# Am I Vulnerable To 'Insufficient Cryptography'?
+
+There are several ways in which insecure cryptography can manifest in a mobile application:
+
+**Weak Encryption Algorithms**: The mobile app may use encryption algorithms that are known to be weak or vulnerable to attacks. These algorithms may have known weaknesses, be outdated, or lack the necessary level of security to protect sensitive data effectively.
+
+**Insufficient Key Length**: Inadequate key length can weaken the encryption strength. If the mobile app uses short or easily guessable encryption keys, it becomes easier for attackers to decrypt the encrypted data through brute-force or other cryptographic attacks.
+
+**Improper Key Management**: Poor key management practices, such as storing encryption keys insecurely or transmitting them in plain text, can expose the keys to unauthorized access. Attackers who gain access to the keys can decrypt the data without difficulty.
+
+**Flawed Encryption Implementation**: The encryption/decryption process itself may be implemented incorrectly or contain programming flaws. These implementation errors can introduce vulnerabilities that attackers can exploit to bypass or weaken the encryption protections.
+
+**Insecure Storage of Encryption Keys**: If the encryption keys are stored insecurely on the mobile device, such as in plain text or in easily accessible locations, attackers with physical or unauthorized access to the device can retrieve the keys and decrypt the protected data.
+
+**Lack of Secure Transport Layer**: When transmitting encrypted data over networks, it is crucial to use secure transport layer protocols like HTTPS. If the mobile app fails to implement secure transport protocols, encrypted data may be vulnerable to interception or tampering during transmission.
```

Comment:
Also lack of encryption when storing critical data, e.g. license keys and similar.